parse/NamedValueRefParser.cpp

```diff
@@ -0,0 +1,131 @@
+         ErrorLogger() << "Registering from named_values.focs.txt : add a AnyValueRef to named_refs map";
+         named_refs.emplace(name, std::move(std::unique_ptr<ValueRef::AnyValueRef>(std::move(ref))));
+         ErrorLogger() << "Registering from named_values.focs.txt : go on...";
+       } else {
+         ::RegisterValueRef(name, std::move(ref_envelope.OpenEnvelope(pass)));
+       }
+    }
+
+    BOOST_PHOENIX_ADAPT_FUNCTION(void, insert_named_ref_, insert_named_ref, 4)
+
+    using start_rule_payload = std::map<std::string, std::unique_ptr<ValueRef::AnyValueRef>>;
+    using start_rule_signature = void(start_rule_payload&);
+
+    struct grammar : public parse::detail::grammar<start_rule_signature> {
+        grammar(const parse::lexer& tok,
+                const std::string& filename,
+                const parse::text_iterator& first,
+                const parse::text_iterator& last) :
+            grammar::base_type(start, "named_value_ref_grammar"),
+            condition_parser(tok, label),
+            string_grammar(tok, label, condition_parser),
+            int_rules(tok, label, condition_parser, string_grammar),
+            double_rules(tok, label, condition_parser, string_grammar)
+        {
+            namespace phoenix = boost::phoenix;
+            namespace qi = boost::spirit::qi;
+
+            qi::_1_type _1;
+            qi::_2_type _2;
+            qi::_3_type _3;
+            qi::_4_type _4;
+            qi::_val_type _val;
+            qi::_pass_type _pass;
+            qi::omit_type omit_;
+            qi::_r1_type _r1;
+
+            named_ref
+               =
+                    ( omit_[tok.Named_]   >> tok.Integer_
+                   > label(tok.Name_) > tok.string
+                   > label(tok.Value_) >  qi::as<parse::detail::MovableEnvelope<ValueRef::ValueRef<int>>>()[int_rules.expr]
+                    ) [ insert_named_ref_(_r1, _2, _3, _pass) ]
+                    |
+                    ( omit_[tok.Named_]   >> tok.Real_
+                   > label(tok.Name_) > tok.string
+                   > label(tok.Value_) >  qi::as<parse::detail::MovableEnvelope<ValueRef::ValueRef<double>>>()[double_rules.expr]
+                    ) [ insert_named_ref_(_r1, _2, _3, _pass) ]
+                ;
+
+
+            start
+              = +named_ref(_r1)
+                ;
+
+            named_ref.name("NamedValueRef");
+
+#if DEBUG_PARSERS
+            debug(named_ref);
+#endif
+
+	    qi::on_error<qi::fail>(start, parse::report_error(filename, first, last, _1, _2, _3, _4));
+        }
+
+        using named_value_ref_rule = parse::detail::rule<
+            void (std::map<std::string, std::unique_ptr<ValueRef::AnyValueRef>>&)>;
+
+        using start_rule = parse::detail::rule<start_rule_signature>;
+
+        parse::detail::Labeller                 label;
+        const parse::conditions_parser_grammar  condition_parser;
```

Comment:
don't think all of these are used / needed...?

Reply:
AFAIU the start_rule is not necessary, but everything else is.